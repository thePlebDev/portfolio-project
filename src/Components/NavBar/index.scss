.loading-enter{
  opacity: 0;

}

.loading-enter-active{
  opacity: 1;
  transition: opacity 5s
}
.loading-enter-done{
  opacity: 1;
}
