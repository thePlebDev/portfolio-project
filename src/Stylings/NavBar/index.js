import styled from 'styled-components';

const BurgerLine1 = styled.div`

  background-color:#0f141e;
  width: 22px;
  height:3px;
  margin:4px 0;

  transition:background-color 1.7s, transform 0.4s;
  transform:${props=>props.state ? 'rotate(-43deg) translate(-5px,6px)' :''};
`
const BurgerLine2 = styled.div`
  transition:background-color 1.7s, transform 0.4s;
  background-color:#0f141e;
  width: 22px;
  height:3px;
  margin:4px 0;
  opacity:${props=>props.state ? 0 :1};
`
const BurgerLine3 = styled(BurgerLine1)`
  transform:${props=>props.state ? 'rotate(47deg) translate(-4px,-4px)' :''};
`

const Container = styled.div`

  padding:12px;
  display:flex;
  margin-top:15px;
  position:relative;
  margin-right:5%;
  position:relative;

`
const Text = styled.div`
  color:#0f141e;
  transition:all 2s;
  z-index:1;
  align-self:center;
  margin-right:25px;
  text-transform:uppercase;
  letter-spacing:0.2em;
  font-weight:700;

`
const MovingText = styled(Text)`
  color:#0f141e;
  opacity:${props=>props.state?'1':'0'};
  margin-top:20px;

`



const BurgerContainer = styled.div`
  margin-right:10px;
  align-self:center;
  position:absolute;
  right:0;
  z-index:1;
  cursor:pointer;

`

export {BurgerLine1,BurgerLine2,BurgerLine3,Container,Text,BurgerContainer,MovingText}
