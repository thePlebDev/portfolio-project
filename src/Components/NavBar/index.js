import React,{useState,useEffect} from 'react';
import styled from 'styled-components'
import { CSSTransition } from 'react-transition-group'

import Hamburger from '../Hamburger'
import {MovingText} from '../../Stylings/NavBar';
import './index.scss'

import useLoading from '../../Hooks/useLoadingAnimation'

const Image = styled.img`
width:100px;
height:50px;
`


const NavBar =({state,setState})=>{
  const {show} = useLoading()

  return(
    <div style={{display:'flex',justifyContent:'space-between',position:'relative'}}>
        <div style={{display:'flex',justifyContent:'center',alignItems:'center',marginLeft:'12%'}}>
            <MovingText state={show}> Elliott Software Development</MovingText>
        </div>
        <Hamburger state={state} setState={setState}/>
    </div>
  )
}

export default NavBar
