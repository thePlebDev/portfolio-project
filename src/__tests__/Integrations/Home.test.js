import React from 'react';
import {render} from "@testing-library/react";
import userEvent from '@testing-library/user-event'


import NavBar from '../../Components/NavBar'


const mockFunc1 = jest.fn()
const mockFunc2 = jest.fn()

Object.defineProperty(window,"sessionStorage",{
  value:{
    getItem:mockFunc1,
    setItem:mockFunc2
  }
})

describe('testing the home components',()=>{
  it('should test the ui upon render',()=>{
    const {asFragment} = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();


  });

  it('should test the ui change',()=>{
    const theme = true;
    const setTheme = jest.fn()
    const {getByTestId} = render(<NavBar theme={theme} setTheme={setTheme}/>);

    let toggle = getByTestId('theme-slider')
    userEvent.click(toggle)
    expect(mockFunc1.mock.calls.length).toBe(1)
    expect(mockFunc2.mock.calls.length).toBe(1)

  })

})
