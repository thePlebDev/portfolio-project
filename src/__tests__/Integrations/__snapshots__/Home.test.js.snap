// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the home components should test the ui upon render 1`] = `
<DocumentFragment>
  <div
    class="sc-dIvqjp hLcnmX"
  >
    <div
      class="sc-hHEjAm jaqisr"
    >
      <h2>
        Tristan Elliott
      </h2>
    </div>
    <div
      class="sc-dlMBXb exGJyY"
    >
      <a
        data-testid="link"
        href="https://www.youtube.com/channel/UCc9o1zG46k4voqOT7C6Ankw"
        rel="noreferrer noopener"
        target="_blank"
      >
        <span
          class="sc-kfYqjs jumAtR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 30px; padding: 10px; cursor: pointer;"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
            />
          </svg>
        </span>
      </a>
      <a
        data-testid="link"
        href="https://twitter.com/Tristanthewebd1"
        rel="noreferrer noopener"
        target="_blank"
      >
        <span
          class="sc-kfYqjs jumAtR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 30px; padding: 10px; cursor: pointer;"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
        </span>
      </a>
      <a
        data-testid="link"
        href="https://github.com/thePlebDev"
        rel="noreferrer noopener"
        target="_blank"
      >
        <span
          class="sc-kfYqjs jumAtR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 30px; padding: 10px; cursor: pointer;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
      </a>
      <div>
        <span
          class="sc-kfYqjs jumAtR"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="search"
            focusable="false"
            style="font-size: 30px; padding: 10px; cursor: pointer;"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
      </div>
      <label
        class="sc-fKgIGh iyneAj"
        for="theme"
      >
        <input
          class="sc-bCwgka eNKQZi"
          data-testid="theme-slider"
          id="theme"
          type="checkbox"
        />
        <span
          class="sc-iwaifL jkAlAa"
        />
      </label>
    </div>
  </div>
</DocumentFragment>
`;
